### PR TITLE
feat(oci): add storage admin delete exclusion check

### LINKED ISSUE #10567: [OCI] Add CIS 3.1 1.15 compliance scans
### Feature search

- [x] I have searched the existing issues and this feature has not been requested yet or is already in our [Public Roadmap](https://roadmap.prowler.com/roadmap)

### Which component would this feature affect?

Prowler CLI/SDK

### Related to specific cloud provider?

Not provider-specific

### New feature motivation

CIS 3.1 Standard has a check 1.15 which is currently not implemented

### Solution Proposed

A new check should be created to detect file-storage service level admins that have not been scoped properly

### Use case and benefits

Clients trying to comply with CIS 3.1 standards will be able to scan for these misconfigurations

### Describe alternatives you've considered

Manually going through policies is a viable alternative

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by HugoPBrito**
### Context

Fix #10567

OCI CIS 3.1 control 1.15 requires storage service-level administrators to manage storage resources without being able to delete the resources they manage. A previous attempt in #10568 was closed after review because it treated any `WHERE` clause as compliant, even when delete permissions were still allowed.

### Description

This PR adds the `identity_storage_service_level_admins_scoped` check for the OCI provider.

The check reviews active OCI IAM policies and verifies storage `manage` statements explicitly exclude the relevant delete permissions with `request.permission`, including:

- Block Volume: `VOLUME_DELETE`, `VOLUME_BACKUP_DELETE`
- File Storage: `FILE_SYSTEM_DELETE`, `MOUNT_TARGET_DELETE`, `EXPORT_SET_DELETE`
- Object Storage: `OBJECT_DELETE`, `BUCKET_DELETE`

The parser is intentionally conservative: generic `WHERE` clauses and disjunctive conditions such as `ANY`, `OR`, or `||` do not produce PASS results unless the required delete exclusion is enforced.

### Steps to review

- Review the check logic in `identity_storage_service_level_admins_scoped.py`.
- Review the CIS 3.1 requirement `1.15` mapping in `cis_3.1_oraclecloud.json`.
- Run the targeted tests:

```bash
PYTHONDONTWRITEBYTECODE=1 uv run pytest -q -p no:cacheprovider tests/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped_test.py
```

- Optional validation commands:

```bash
uv run python -m json.tool prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.metadata.json >/dev/null
uv run python -m json.tool prowler/compliance/oraclecloud/cis_3.1_oraclecloud.json >/dev/null
```

### Checklist

<details>

<summary><b>Community Checklist</b></summary>

- [x] This feature/issue is listed in [here](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or roadmap.prowler.com
- [ ] Is it assigned to me, if not, request it via the issue/feature in [here](https://github.com/prowler-cloud/prowler/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or [Prowler Community Slack](goto.prowler.com/slack)

</details>

- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.
- [x] Review if is needed to change the [Readme.md](https://github.com/prowler-cloud/prowler/blob/master/README.md)
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/prowler/CHANGELOG.md), if applicable.

#### SDK/CLI
- Are there new checks included in this PR? Yes
    - If so, do we need to update permissions for the provider? Please review this carefully. The check uses existing OCI Identity policy collection.

#### UI
- [x] All issue/task requirements work as expected on the UI. Not applicable, SDK-only change.
- [x] If this PR adds or updates npm dependencies, include package-health evidence (maintenance, popularity, known vulnerabilities, license, release age) and explain why existing/native alternatives are insufficient. Not applicable.
- [x] Screenshots/Video of the functionality flow (if applicable) - Mobile (X < 640px). Not applicable.
- [x] Screenshots/Video of the functionality flow (if applicable) - Table (640px > X < 1024px). Not applicable.
- [x] Screenshots/Video of the functionality flow (if applicable) - Desktop (X > 1024px). Not applicable.
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/ui/CHANGELOG.md), if applicable. Not applicable.

#### API
- [x] All issue/task requirements work as expected on the API. Not applicable, SDK-only change.
- [x] Endpoint response output (if applicable). Not applicable.
- [x] EXPLAIN ANALYZE output for new/modified queries or indexes (if applicable). Not applicable.
- [x] Performance test results (if applicable). Not applicable.
- [x] Any other relevant evidence of the implementation (if applicable). Targeted pytest and JSON validation commands are listed above.
- [x] Verify if API specs need to be regenerated. Not applicable.
- [x] Check if version updates are required (e.g., specs, uv, etc.). Not applicable.
- [x] Ensure new entries are added to [CHANGELOG.md](https://github.com/prowler-cloud/prowler/blob/master/api/CHANGELOG.md), if applicable. Not applicable.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

## Summary by CodeRabbit

* **New Features**
  * Added an Oracle Cloud Identity check to ensure storage service-level admin policies explicitly exclude delete permissions to enforce separation of duties.

* **Documentation**
  * Changelog and guidance updated with control mapping, remediation steps (CLI/Console/Terraform examples), and recommended policy patterns.

* **Tests**
  * New test suite validating PASS/FAIL behavior across policy scenarios, resource families, parsing edge cases, and ignored/inactive policy cases.

### CODE REVIEW

File: prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    return " ".join(statement.strip().split())
+
+
+def _has_disjunctive_condition(statement: str) -> bool:
+    """Return True when the WHERE condition can allow alternate branches."""
+    condition = re.split(r"\bwhere\b", statement, flags=re.IGNORECASE, maxsplit=1)
+    if len(condition) != 2:
+        return False
+
+    return bool(re.search(r"\b(any|or)\b|\|\|", condition[1], re.IGNORECASE))
```
Comment (danibarranqueroo) [inline prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py]: The disjunction guard searches for \b(any|or)\b|\|\| anywhere after where, including inside quoted values. That produces a false‑positive FAIL on legitimate policies whose tag namespaces/values happen to contain those tokens:

Allow group BucketUsers to manage buckets in tenancy where request.permission!='BUCKET_DELETE' and target.tag.namespace='any-tag'

Delete is properly excluded, but the statement is flagged as disjunctive and fails.

Could we strip quoted literals before scanning, or limit the check to boolean operators outside '…' / "…"?

File: prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            if violations:
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"Policy '{policy.name}' allows storage service-level administrators to manage storage resources without explicitly excluding required delete permissions: "
+                    + "; ".join(violations)
+                    + "."
+                )
+            else:
+                report.status = "PASS"
+                report.status_extended = f"Policy '{policy.name}' excludes required storage delete permissions from storage manage statements."
```
Comment (danibarranqueroo) [inline prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py:154]: Every active non‑tenant‑admin policy currently gets a finding, even when it has no storage manage statements at all. For those the message reads:

> "Policy 'X' excludes required storage delete permissions from storage manage statements."


which isn't accurate — there are no such statements to exclude anything from — and it'll produce a lot of noisy PASS findings in real tenancies. Two options:

1. Only emit a per-policy finding when the policy contains at least one storage manage statement (skip the rest entirely, and rely on the "no findings" tenancy-level PASS).
2. Keep the per-policy report but reword to something neutrally true, e.g. "Policy 'X' does not grant storage admin manage permissions without excluding delete permissions."

File: prowler/CHANGELOG.md
Comment (danibarranqueroo) [inline prowler/CHANGELOG.md:1]: Update this with master and move the entry to 5.31.0

**Review by danibarranqueroo — CHANGES_REQUESTED** [review]
Good job! I left some small comments 💯

File: prowler/CHANGELOG.md
Comment (HugoPBrito) [reply prowler/CHANGELOG.md:1]: Done, thanks!

File: prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py (hunk 2 above)
Comment (HugoPBrito) [reply prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py:154]: Done, thanks!

File: prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py (hunk 1 above)
Comment (HugoPBrito) [reply prowler/providers/oraclecloud/services/identity/identity_storage_service_level_admins_scoped/identity_storage_service_level_admins_scoped.py]: Thanks!

**Review by danibarranqueroo — APPROVED** [review]

**Review by pedrooot — APPROVED** [review]
